feat: reference for the self referencing view error (#37560)
supabase/supabase: apps/docs/content/guides/platform/backups.mdx

- validating against the current time, e.g. that the row being inserted references a future event
- validating the contents of a row against the contents of another table

#### Views that reference themselves

> Views that directly or indirectly reference themselves can cause **cyclic dependency errors** during a logical backup restore. These views are also **invalid and unusable** , any query against them will result in an error at runtime.

Reviewer: ```suggestion
Views that directly or indirectly reference themselves will cause logical restores to fail due to cyclic dependency errors. These views are also invalid and unusable in Postgres, and any query against them will result in a runtime error.
```

stylistically we don't seem to use bold text in this fashion in the rest of the doc. Not a big deal if you want to retain it though.

can -> will
logical backup restore -> logical restore (consistent with doc)

**Example:**

```
-- Direct self-reference
CREATE VIEW my_view AS
  SELECT * FROM my_view;

-- Indirect circular reference
CREATE VIEW v1 AS SELECT * FROM v2;
CREATE VIEW v2 AS SELECT * FROM v1;
```

> -- Drop the offending view.

Reviewer: stray comment? Can make it a full sentence e.g. `Drop the offending view from your database, or delete them from the logical backup to make it restorable.`
